File: 2210-count-hills-and-valleys-in-an-array/2210-count-hills-and-valleys-in-an-array.py
class Solution:
    def countHillValley(self, v):
        n = len(v)
        a=[]
        for i in range (0,n-1):
            if v[i] != v[i+1]:
                a.append(v[i])
        ans = 0
        a.append(v[n-1])
        for i in range (1,len(a)-1):
            if a[i]>a[i-1] and a[i]>a[i+1] or a[i]<a[i-1] and a[i]<a[i+1]:
                ans+=1
        return ans
        

File: 2210-count-hills-and-valleys-in-an-array/2210-count-hills-and-valleys-in-an-array.cpp
class Solution {
public:
    int countHillValley(vector<int>& v) {
        int n=v.size();
        vector<int> a;
        for(int i=0;i<n-1;i++)
        {
            if(v[i]!=v[i+1]) a.push_back(v[i]);
        }
        a.push_back(v[n-1]);
        int ans=0;
        for(int i=1;i<a.size()-1;i++)
        {
            if(a[i]>a[i-1] && a[i]>a[i+1] || a[i]<a[i-1] && a[i]<a[i+1]) ans++;
        }
        return ans;

        
    }
};